wallet/src/ui/app/hooks/useMutationObserver.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { useEffect, useState } from 'react';
+
+import type { RefObject } from 'react';
+
+interface Props {
+    target: RefObject<Element> | Element | Node | null;
+    options: MutationObserverInit;
+    callback: MutationCallback;
+}
+
+const getRefElement = <T>(
+    element?: RefObject<Element> | T
+): Element | T | undefined | null => {
+    if (element && 'current' in element) {
+        return element.current;
+    }
+
+    return element;
+};
+
+const useMutationObserver = ({
+    target,
+    options = {},
+    callback,
+}: Props): void => {
+    const [observer, setObserver] = useState<MutationObserver | null>(null);
```

Comment:
You can use a memo to get this instead of an effect which will improve perf due to a saved render:

```typescript
const observer = useMemo(() => new MutationObserver(callback)), [callback]);
```